### DISCUSSION TITLE
Tailwind + NextJS = background-image not shown

### DISCUSSION THREAD

**Discussion by diegoalvarezmgl** [tailwindlabs/tailwindcss#6918] [Help]
Hello. I am coding a "switch" component which works as a toggle. When I click on it, it changes its state and it should change its background image.

my styles/global.css:

```
#toggle-switch::after {
  content: "";
  @apply h-5 w-5 rounded-full absolute scale-110 transform transition duration-200 bg-myfirstimage;
}

#toggle-switch:checked::after {
  @apply scale-110 transform translate-x-5 bg-mysecondimage;
}
```

And my tailwind.config.js:

```
theme: {
  extend: {
      backgroundImage: (theme) => ({
        myfirstimage: "url('/img/myfirstimage.svg')",
        mysecondimage: "url('/img/mysecondimage.svg')",
      }),
  }
}
```

When I hover on my "bg-myfirstimage" I see properly configured it like:

```
.bg-myfirstimage {
       background-image: url('/img/myfirstimage.svg');
}
```

However, that bg does not appear on my component (and of course, if I change it for bg-black i.e. it appears black, correctly).

I tried changing the folder "img" to public, and it is the same, it doesn't seem to take the .svg.

So, how can I make Talwind to take the .svg? Could anyone help me? Thanks!!

**Comment by kingsley-sunny** [comment]
not only svg, it does not works for png, jpg, jpeg files i think it is a bug

**Comment by sebastiandine** [comment]
Facing the same issue right now

**Reply by sebastiandine** [reply]
I managed the solve this as follows: What I found out is, that `url()` seems to refer to the `public/` directory of the project when the url starts from "root" (e.g. `url(/herobg.png)` refers to `public/herbg.png`). If you want to refer an image from another directory other than `public/`, you need to "step out" of the `public` directory first (e.g. `url(../assets/herobg.png)` refers to `assets/herobg.png`).

**Comment by andrepichardo** [comment]
Hi!

Try adding the bg class as an arbitrary value like this to check if that works:

```
bg-[url('/img/myfirstimage.svg')]
```

If not, it's probably a bug. 
Hope it helps!

**Reply by satishkumarsajjan** [reply]
Did it work for any one?
it is not working for me.

**Reply by Zehad-10** [reply]
```
 theme: {
        extend: {
            backgroundImage: {
                your_image: "url('/media/pngs/your_image.png')"
            }
        }
    },
    
    try like this, note:: `/media` is located inside a public folder

**Reply by AbdoMagdodz** [reply]
Thanks @Zehad-10 this worked for me

**Comment by lautaro-arias** [comment]
la respuesta es :  theme: {
    extend: {
      backgroundImage: {
        'bgPage':"url('/assets/bgPage.jpg')",
      },

tengo en mi carpeta public > assets > bgPage  

bg-[url('/assets/bgPage.jpg')]   si no tenes la carpeta assets te queda : "url('/bgPage.jpg')"
